controllers/CommentController.js

```diff
@@ -137,4 +138,78 @@ export default class CommentController {
       return next(err);
     }
   }
+
+  /**
+   * Method to edit a comment
+   *
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} details of the edited comment
+   * @memberof CommentController
+   */
+  static async editComment(req, res, next) {
+    try {
+      const { id } = req.user;
+      const { commentId } = req.params;
+      const { body } = req.body;
+      const comment = await Comment.findOne({ where: { id: commentId } });
+      if (!comment) {
+        return errorResponse(res, 404, { message: 'Comment does not exist' });
+      }
+      if (comment.dataValues.userId !== id) {
+        return errorResponse(res, 403, 'You are not allowed to edit another user\'s comment');
+      }
+      await CommentHistory.create({
+        userId: id,
+        commentId,
+        oldComment: comment.body,
+      });
+      const editedComment = await Comment.update(
+        {
+          body
+        },
+        {
+          where: { id: commentId },
+          returning: true
+        }
+      );
+      return successResponse(res, 200, 'editedComment', editedComment[1][0]);
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * Method to get a comment edit history
+   *
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} details of the edited comment
+   * @memberof CommentController
+   */
+  static async getEditCommentHistory(req, res, next) {
+    try {
+      const { id } = req.user;
+      const { commentId } = req.params;
+      const commentHistory = await CommentHistory.findAll({
+        where: { commentId },
+        order: [
+          ['updatedAt', 'DESC'],
+        ],
+      });
+      if (!commentHistory[0]) {
+        return errorResponse(res, 404, { message: 'Comment does not have edit history' });
+      }
+      if (commentHistory[0].dataValues.userId !== id) {
+        return errorResponse(res, 403, 'You are not allowed to edit another user\'s comment');
```

Comment:
@OdunayoOkebunmi  Is this error message correct? Is this not meant to be 'You are not allowed to view another user\'s comment history'